op-node/rollup/driver/sequencer.go

```diff
@@ -230,6 +249,11 @@ func (d *Sequencer) RunNextSequencerAction(ctx context.Context) (*eth.ExecutionP
 			return payload, nil
 		}
 	} else {
+		if err := d.CanStartBuildingBlock(); err != nil {
+			d.log.Error("sequencer is not allowed to build block", "err", err)
+			d.nextAction = d.timeNow().Add(time.Second)
+			return nil, err
+		}
```

Comment:
Hi @bnoieh, thanks for your review.

1/2. I agree with your comments on (1) and (2). If we move the check `coordinator_startBuildingBlock` to the front of `case <- sequencerCh`, we can achieve the behavior of "inactive sequencer not entering case <- sequencerCh" and cancel all sequencer actions. This suggestion is great, and I will update it as soon as possible.
3. The change is related to 1/2. I will make some experiments and then reply.
4. `admin_sequencerActive` should return `false` for inactive sequencers.
5. It's a big question. In simple terms, the current HA mechanism cannot ensure to prevent fork 100% effectively.